src/bigquery/base.py

```diff
@@ -0,0 +1,145 @@
+# Copyright © 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Dharmendra G Patel <dhpatel@redhat.com>
+#
+"""Implementation Bigquery builder base."""
+import os
+import json
+import time
+import tempfile
+from rudra import logger
+from src.config.settings import GCP_SETTINGS
+from google.cloud import bigquery
+
+_POLLING_DELAY = 1  # sec
+
+
+class BigqueryBuilder:
+    """BigqueryBuilder class Implementation."""
+
+    def __init__(self, query_job_config=None):
+        """Initialize the BigqueryBuilder object."""
+        logger.info('Storing BigQuery Auth Credentials')
+        key_file_contents = {
+            "type": GCP_SETTINGS.type,
+            "project_id": GCP_SETTINGS.project_id,
+            "private_key_id": GCP_SETTINGS.private_key_id,
+            "private_key": GCP_SETTINGS.private_key,
+            "client_email": GCP_SETTINGS.client_email,
+            "client_id": GCP_SETTINGS.client_id,
+            "auth_uri": GCP_SETTINGS.auth_uri,
+            "token_uri": GCP_SETTINGS.token_uri,
+            "auth_provider_x509_cert_url": GCP_SETTINGS.auth_provider_x509_cert_url,
+            "client_x509_cert_url": GCP_SETTINGS.client_x509_cert_url,
+        }
+        tfile = tempfile.NamedTemporaryFile(mode='w+', delete=True)
+        tfile.write(json.dumps(key_file_contents))
+        tfile.flush()
+        tfile.seek(0)
+        self.credential_path = tfile.name
```

Comment:
You can mount the file as secrete. Refer the following template which mounts the same bigquery file
https://github.com/kubesecurity/openshift-probable-vulnerabilities/blob/master/deployment/template-cronjob.yaml#L91

Reply:
This is temporary file we need during authentication only, we remove this file in the same flow as soon as we initialise google cloud client.

Reply:
Nope, this is not a correct way to mount a secret which is in the form of file.

Reply:
This is the current way this pipeline was configure. We can modify and re-arrange all secrete as suggested. This will bring yet another changes in deployment template and secrete structure. Adding more scope to this bulky PR is what I want to avoid. I have raised a separate ticket to more all GCP secretes from ENV to a file => https://issues.redhat.com/browse/APPAI-1823